api/src/main/java/edu/wpi/first/shuffleboard/api/data/DataTypes.java

```diff
@@ -86,6 +61,104 @@ public static void register(DataType<?> dataType) {
   }
 
   /**
+   * Gets the data type most relevant to a Java class.
+   */
+  @SuppressWarnings("unchecked")
+  public static <T> Optional<DataType<T>> forJavaType(Class<T> type) {
+    if (type.isPrimitive()) {
+      return forJavaType((Class) boxedType(type));
+    }
+    return (Optional) typeCache.computeIfAbsent(type, __ -> {
+      if (DataType.class.isAssignableFrom(type)) {
+        return forType((Class<DataType>) type);
+      }
+      Optional<DataType> naive = dataTypes.values().stream()
+          .filter(t -> type.equals(t.getJavaClass()))
+          .findAny();
+      if (naive.isPresent()) {
+        return naive;
+      } else {
+        // Check Java class hierarchy
+        Comparator<Class<?>> classComparator = closestTo(type);
+        List<DataType> sorted = dataTypes.values().stream()
+            .filter(t -> t != All)
+            .sorted((t1, t2) -> classComparator.reversed().compare(t1.getJavaClass(), t2.getJavaClass()))
+            .collect(Collectors.toList());
+        return Optional.of(sorted.get(0));
+      }
+    });
+  }
+
+  private static Comparator<Class<?>> closestTo(Class<?> target) {
+    return (o1, o2) -> {
+      if (o1 == o2) {
+        return 0;
+      }
+      if (o1 == null) {
+        return -1;
+      } else if (o2 == null) {
+        return 1;
+      }
+      boolean isO1Superclass = o1.isAssignableFrom(target);
+      boolean isO2Superclass = o2.isAssignableFrom(target);
+      if (isO1Superclass && !isO2Superclass) {
+        return 1;
+      } else if (!isO1Superclass && isO2Superclass) {
+        return -1;
+      } else if (!isO1Superclass) { //NOPMD
+        // Neither is a superclass; order doesn't matter
+        return 0;
+      } else {
+        // Target inherits from both
+        int c1 = 0;
+        int c2 = 0;
+        Class sup = o1.getSuperclass();
+        while (sup != Object.class && sup != null) {
+          sup = sup.getSuperclass();
+          c1++;
+        }
+        sup = o2.getSuperclass();
+        while (sup != Object.class && sup != null) {
+          sup = sup.getSuperclass();
+          c2++;
+        }
+        return Integer.compare(c1, c2);
+      }
+    };
+  }
+
+  public static void clearCache() {
+    typeCache.clear();
+  }
+
+  /**
+   * A.
+   */
+  public static Set<DataType> forJavaTypes(Class<?>... types) {
+    return Stream.of(types)
+        .map(DataTypes::forJavaType)
+        .filter(Optional::isPresent)
+        .map(Optional::get)
+        .collect(Collectors.toSet());
+  }
+
+  private static Class<?> boxedType(Class<?> primitiveType) {
```

Comment:
This could be replaced with `Primitives.wrap`:

http://google.github.io/guava/releases/21.0/api/docs/com/google/common/primitives/Primitives.html#wrap-java.lang.Class-